### DISCUSSION TITLE
Jupyter Kernel Crashes on Windows

### DISCUSSION THREAD

**Discussion by JessicaJohnBritto** [block-hczhai/block2-preview#118] [Q&A]
I am trying to plot the first excitation gap as a function of the Hopping Amplitude J. With increase in J value, it takes more time to compute the gap using the script given here ([script-1](https://github.com/JessicaJohnBritto/MTP/blob/main/Initial%20Codes/Block2_DMRG_final-Copy1.ipynb)). For example, if t goes from 0 to 1 with a step of 0.1, after computing gap for these cases, the kernel crashes. 

To debug this, I run the script on python interpreter and got this error
![image](https://github.com/user-attachments/assets/16bba144-dccf-40d7-b3c9-b32c33eaa365)
I tried uninstalling and reinstalling numpy and matplotlib as given some of the stackexchange answers - these did not work either.

I tested the same using the script ([script-2](https://github.com/JessicaJohnBritto/MTP/blob/main/Initial%20Codes/Block2_DMRG_final.ipynb)) for Bose Hubbard Model on the documentation site. In this case, it takes about approx 60 seconds for each value of J to compute the gap.

Also, the two scripts above when on run Mac, it gave the output in a few seconds.

**Comment by hczhai** [comment]
> Also, the two scripts above when on run Mac, it gave the output in a few seconds.

It is common that when you run a code on different hardware, you get different speed. If it is more stable on Mac, you may simply use Mac.

> For example, if t goes from 0 to 1 with a step of 0.1, after computing gap for these cases, the kernel crashes.

If you make significant changes to what you get from the documentation, it is your responsibility to understand its behavior. In general I do not recommend you to reorganize the documentation script and then assume the outcome will still make sense. In fact, there are some evident errors in your script, such as:

```python3
Q, driver, b = initialVals(params)
gaps = []
for T in params['t_list']:
    b.add_term("CD", np.array([[i, i+1, i+1, i] for i in range(params['L']-1)]).flatten(), -T)
```

Here when you keep calling ``b.add_term`` inside the loop without creating a new ``b``, your Hamiltonian will have T=0.1 terms in the second loop, and both T=0.1 and T=0.2 terms in the third loop, etc, which is wrong.

The safest way is to run the 10 cases in 10 separate scripts, and each script should look exactly like the one given in the documentation, apart from some small changes for the parameters.

**Reply by JessicaJohnBritto** [reply]
Oh okay, got it. Thanks

**Reply by JessicaJohnBritto** [reply]
The performance improved a lot after removing these files `libiomp5md.dll` and `libiomp5md_db.dll` from anaconda/../envs/XY/Library/bin.

The error was resolved after using this command `conda install nomkl --channel conda-forge` inside the virtual environment.
